lib/vanagon/project.rb

```diff
@@ -101,6 +102,17 @@ def get_replaces
       replaces.flatten.uniq
     end
 
+    # Collects all of the conflicts for the project and its components
+    def get_conflicts
+      conflicts = []
+      # Including replaces in the conflicts list helps to ensure
+      # compatibility
+      conflicts.push get_replaces
+      conflicts.push @conflicts.flatten
+      conflicts.push @components.map(&:conflicts).flatten
+      conflicts.flatten.uniq
```

Comment:
For further clarification (since we're working asynchronously), I'd **really** like it if we used `Set` instead of `Array` for things that shouldn't allow duplicate entries (like `Conflicts`, `Provides`, `Obsoletes`, `Replaces`, etc.) but I'm not going to propose we gut this wholesale for the sake of this PR. While this probably follows what we've done for other methods, I'd like to see less use of intermediate variable manipulation if possible. Maybe something like this?

```ruby
def get_conflicts
  # Including get_replaces in the conflicts list to provide backwards compatibility
  # with previous behavior
  conflicts = @components.flat_map(&:conflicts) + @conflicts + get_replaces
  # Mash the whole thing down into a flat Array and run uniq to simplify the list
  conflicts.flatten.uniq
end
```

In my quick and dirty testing, that looks like this when it's used:

```ruby
>> a = %w(tar gnutar gtar bsdtar)
=> [
    [0] "tar",
    [1] "gnutar",
    [2] "gtar",
    [3] "bsdtar"
]
>> b = [%w(nestar nectar), 'tar', 'gtar']
=> [
    [0] [
        [0] "nestar",
        [1] "nectar"
    ],
    [1] "tar",
    [2] "gtar"
]
>> c = OpenStruct.new(conflicts: %w(btar ctar allstar))
=> #<OpenStruct conflicts=["btar", "ctar", "allstar"]>
>> d = OpenStruct.new(conflicts: %w(redstar whitestar tar))
=> #<OpenStruct conflicts=["redstar", "whitestar", "tar"]>
>> @components = [c, d]
=> [
    [0] #<OpenStruct conflicts=["btar", "ctar", "allstar"]>,
    [1] #<OpenStruct conflicts=["redstar", "whitestar", "tar"]>
]
>> conflicts = @components.flat_map(&:conflicts) + a + b
=> [
    [ 0] "btar",
    [ 1] "ctar",
    [ 2] "allstar",
    [ 3] "redstar",
    [ 4] "whitestar",
    [ 5] "tar",
    [ 6] "tar",
    [ 7] "gnutar",
    [ 8] "gtar",
    [ 9] "bsdtar",
    [10] [
        [0] "nestar",
        [1] "nectar"
    ],
    [11] "tar",
    [12] "gtar"
]
>> conflicts.flatten.uniq
=> [
    [ 0] "btar",
    [ 1] "ctar",
    [ 2] "allstar",
    [ 3] "redstar",
    [ 4] "whitestar",
    [ 5] "tar",
    [ 6] "gnutar",
    [ 7] "gtar",
    [ 8] "bsdtar",
    [ 9] "nestar",
    [10] "nectar"
]
>>
```